src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/scheduler/RCAScheduler.java

```diff
@@ -99,52 +93,30 @@ public void start() {
     createExecutorPools();
 
     if (scheduledPool != null && role != NodeRole.UNKNOWN) {
-      futureHandle =
-          scheduledPool.scheduleAtFixedRate(
-              new RCASchedulerTask(
-                  10000, rcaSchedulerPeriodicExecutor, connectedComponents, db, persistable, rcaConf, net),
-              1,
-              PERIODICITY_SECONDS,
-              TimeUnit.SECONDS);
-      startExceptionHandlerThread();
       schedulerState = RcaSchedulerState.STATE_STARTED;
+
+      final RCASchedulerTask task = new RCASchedulerTask(
+          10000, rcaSchedulerPeriodicExecutor, connectedComponents, db, persistable, rcaConf, net);
+      while (schedulerState == RcaSchedulerState.STATE_STARTED) {
+        try {
+          long startTime = System.currentTimeMillis();
+          task.run();
+          long duration = System.currentTimeMillis() - startTime;
+          if (duration < PERIODICITY_IN_MS) {
+            Thread.sleep(PERIODICITY_IN_MS - duration);
+          }
+        } catch (InterruptedException ie) {
+          LOG.error("Rca scheduler thread sleep interrupted. Reason: {}", ie.getMessage());
+          LOG.error(ie);
+          shutdown();
+          schedulerState = RcaSchedulerState.STATE_STOPPED_DUE_TO_EXCEPTION;
```

Comment:
We are catching exception here. How are we reporting that a thread shutdown because of exception in the stats collector ?